hs_core/models.py

```diff
@@ -4575,18 +4589,43 @@ def create_element(self, element_model_name, **kwargs):
         """Create any supported metadata element."""
         model_type = self._get_metadata_element_model_type(element_model_name)
         kwargs['content_object'] = self
+        element_model_name = element_model_name.lower()
+        if self.resource.raccess.published:
```

Comment:
This is the first time access control constraints have been included in the create/update/delete element methods.  I see why you did it this way but I'm wondering if there is a better place for it higher up the stack.

Reply:
https://github.com/hydroshare/hydroshare/pull/3969/files#r609913930

Reply:
The only higher up place I can think of is the metadata editing view function(s). However, the view functions are different for the REST call and front-end calls. Since the metadata editing constraints are for the published resource I feel these constraints need be down the stack as much as possible.

Reply:
>  in `create_element()`, `update_element()`, and `delete`

You're right, I misread it.